### PR TITLE
Basic theme support

### PR DESCRIPTION
**Pull request by baconglucose**
# Theme switching

The user interface is not yet implemented.
Change theme like this in the browser console once the page has loaded:
```
changeTheme('theme1')
changeTheme('theme2')
changeTheme('theme3')
changeTheme('standard')
```

New themes can easily be defined now!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1 @@
 .htaccess
-README.md
```
Comment (baconglucose) [inline .gitignore:2]: The readme can be on github :)

File: content/index.html [hunk 2]
```diff
@@ -19,7 +20,7 @@ <h1 class=MainTitle id=titleTop></h1>
                 </div>
             </section>
             <section>
-                <a class="weatherwidget-io" href="https://forecast7.com/de/46d957d45/bern/" data-label_1="Bärn" data-label_2="Wätter" data-theme="dark" >Bärn Wätter</a>
+                <a class="weatherwidget-io" href="https://forecast7.com/de/46d957d45/bern/" data-label_1="Bärn" data-label_2="Wätter" data-icons="" data-theme="dark" data-font="Ubuntu" data-basecolor="#000000" data-textcolor="#FFFFFF">Bärn Wätter</a>
```
Comment (baconglucose) [inline content/index.html]: Here I added color properties and ubuntu font to the widget.

File: content/script/code.js [hunk 3]
```diff
@@ -110,4 +109,22 @@ function timeInDialect() {
...
+        document.querySelector('html').classList.remove(themeList[style]);
+    }
+    if (theme != 'standard') {
+        document.querySelector('html').classList.add(theme);
+    }
+
+    // This part updates the weatherwidget attributes and reinitializes the widget.
+    document.querySelector('a').setAttribute('data-basecolor', getComputedStyle(document.documentElement).getPropertyValue('--color1'));
+    document.querySelector('a').setAttribute('data-textcolor', getComputedStyle(document.documentElement).getPropertyValue('--color0'));
+    __weatherwidget_init()
```
Comment (baconglucose) [inline content/script/code.js:129]: This function can initialize the widget. We change the color properties of the widget using the theme colors and then reinitialize it.
It is hacky but it works.

File: content/style/main.css [hunk 4]
```diff
@@ -1,25 +1,60 @@
...
+    --color1: black;
+    --font1: 'Ubuntu', sans-serif;
+}
+
+/* Changeable colors */
+.theme1 {
+    /* Refined dark theme */
+    --color0: #D8DEE9;
+    --color1: #434C5E;
+}
```
Comment (baconglucose) [inline content/style/main.css:16]: This is how easily a theme can now be defined! Simply add more.

File: .gitignore (hunk 1 above)
Comment (thedcpat) [reply .gitignore:2]: Wasn't intentional. Naturaly the readme should be on github ;)

File: content/script/code.js (hunk 3 above)
Comment (thedcpat) [reply content/script/code.js:129]: Why do you use querySelectors? Wouldn't selecting by id be more accurate and future-proof?

**Review by thedcpat — CHANGES_REQUESTED** [review]
Wouldn't it be more efficient to match the id int the new function in the js code? In my eyes this would also proof itself to be more future proof. Give me feedback on this one please.
The themes are pretty neatly done! Great work.

**Review by thedcpat — APPROVED** [review]

**Pull request merged**